core/modification/src/main/java/datawave/modification/cache/ModificationCache.java

```diff
@@ -0,0 +1,120 @@
+
+    private static final Text MODIFICATION_COLUMN = new Text("m");
+
+    private Map<String,Set<String>> cache = new HashMap<>();
+
+    private final AccumuloConnectionFactory connectionFactory;
+
+    private ModificationConfiguration modificationConfiguration;
+
+    public ModificationCache(AccumuloConnectionFactory connectionFactory, ModificationConfiguration modificationConfiguration) {
+        this.connectionFactory = connectionFactory;
+        this.modificationConfiguration = modificationConfiguration;
+        if (modificationConfiguration != null) {
+            reloadMutableFieldCache();
+        } else {
+            log.error("modificationConfiguration was null");
+        }
+    }
+
+    /**
+     * Reload the cache
+     */
+    public void reloadMutableFieldCache() {
+        Map<String,Set<String>> cache = new HashMap<>();
+        AccumuloClient client = null;
+        BatchScanner s = null;
+        try {
+            Map<String,String> trackingMap = connectionFactory.getTrackingMap(Thread.currentThread().getStackTrace());
+            log.trace("getting mutable list from table " + this.modificationConfiguration.getTableName());
+            log.trace("modificationConfiguration.getPoolName() = " + modificationConfiguration.getPoolName());
+            client = connectionFactory.getClient(null, null, modificationConfiguration.getPoolName(), Priority.ADMIN, trackingMap);
+            log.trace("got connection");
+            s = ScannerHelper.createBatchScanner(client, this.modificationConfiguration.getTableName(),
+                            Collections.singleton(client.securityOperations().getUserAuthorizations(client.whoami())), 8);
+            s.setRanges(Collections.singleton(new Range()));
+            s.fetchColumnFamily(MODIFICATION_COLUMN);
+            for (Entry<Key,Value> e : s) {
+                // Field name is in the row and datatype is in the colq.
+                String datatype = e.getKey().getColumnQualifier().toString();
+                log.trace("datatype = " + datatype);
+                String fieldName = e.getKey().getRow().toString();
+                log.trace("fieldname = " + fieldName);
+                if (null == cache.get(datatype))
+                    cache.put(datatype, new HashSet<>());
+                cache.get(datatype).add(fieldName);
+            }
+            log.trace("cache size = " + cache.size());
+            for (Entry<String,Set<String>> e : cache.entrySet()) {
+                log.trace("datatype = " + e.getKey() + ", fieldcount = " + e.getValue().size());
+            }
+            // now atomically replace the cache
+            this.cache = cache;
+        } catch (Exception e) {
+            log.error("Error during initialization of ModificationCacheBean", e);
+            throw new RuntimeException("Error during initialization of ModificationCacheBean", e);
+        } finally {
+            if (null != s)
+                s.close();
```

Comment:
if s.close() can throw an exception it would be possible to leak the connection

Reply:
FWIW this code was lifted, unchanged from ModificationCacheBean.  This was just an attempt to strip the core functionality from the webservices module, and put it in a common place.